server-database/helpers.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/jackc/pgx/v4"
+)
+
+func ConnectToDB(DB_URL string) *pgx.Conn {
+	conn, err := pgx.Connect(context.Background(), DB_URL)
+
+	if err != nil {
+		log.Fatalf("Unable to connect to database: %s", err.Error())
```

Comment:
In general we don't `log.Fatal` outside of the `main` function itself, and instead return an `error` up which the `main` function can decide how to handle.

One of the biggest reasons for this is that if we end up moving around code so that e.g. we end up connecting to the DB in a request handler or similar, a database going down may cause our whole process to crash. If we're connecting in our `main` function, maybe that's what we want to do, but in a request handler, almost certainly we want to do something like return a 500 rather than crash.

Return an `error` lets us decide how to respond; `log.Fatal` both forces a response, and also means we need to read a lot more code to understand what happens on failure (in general not returning an error implies "nothing can go wrong") - returning an error both allows and forces the caller to think about how to handle the error.

Reply:
Thanks, updated now